src/librustdoc/clean/mod.rs

```diff
@@ -1792,9 +1776,12 @@ impl Clean<Vec<Item>> for (&hir::Item<'_>, Option<Symbol>) {
                 ItemKind::Fn(ref sig, ref generics, body_id) => {
                     clean_fn_or_proc_macro(item, sig, generics, body_id, &mut name, cx)
                 }
-                ItemKind::Macro(ref macro_def) => MacroItem(Macro {
-                    source: display_macro_source(cx, name, macro_def, def_id, item.vis),
```

Comment:
Does this make it possible to remove `item.vis`? Or is it used somewhere else?

Reply:
Interesting question; I'll look into it.

Reply:
In this case, `item` refers to a HIR item if I'm not mistaken. So I assume you're referring to `clean::Item.visibility`, in which case it should likely be possible to remove `clean::Visibility` and compute visibility. I may look into that, but it might be worth saving for a future PR.

Reply:
I've opened #90852 to track this cleanup.

Reply:
Sounds great, thanks!